qualtran/bloqs/rotations/phase_gradient.py

```diff
@@ -183,21 +197,28 @@ def scaled_val(self, x: int) -> int:
         x_fxp = _fxp(x / 2**x_width, x_width).like(_fxp(0, self.phase_bitsize)).astype(float)
         return int(x_fxp.astype(float) * 2**self.phase_bitsize)
 
-    def apply(self, x: int, phase_grad: int) -> Union[int, Iterable[int]]:
-        out = self.on_classical_vals(x=x, phase_grad=phase_grad)
-        return out['x'], out['phase_grad']
+    def apply(self, ctrl: int, x: int, phase_grad: int) -> Union[int, Iterable[int]]:
```

Comment:
You'll have to change this to `*args` because `apply` will be called with only 2 parameters with `self.control` is False.